### ISSUE TITLE
Search for number fields with a given completion

### ISSUE THREAD

**Issue by roed314** [LMFDB/lmfdb#4537]
It would be nice to search for number fields that had a specific local field as one of their completions.

Similarly, on a local field, it would be nice to have a link to this search results page in their related objects.

**Comment by jwj61** [comment]
This looks doable.  The local algebras would have to move from the extras table to the main one.  In doing that, the format used for storing the data can change to facilitate the search.  For a new format, I am thinking of a list of p-adic field labels (so an array of strings).  Then the search can be a test of whether the given label is in the array.  Being an array, some labels may appear more than once to deal with multiplicity, but that's ok.

For a user search, we can start with a search by label; it is the easiest to implement.  Later we can think about allowing polynomials and testing them (does sage have Panayi's algorithm yet?).

**Comment by JohnCremona** [comment]
@jwj61 Woring with elliptic curves was made a lot simpler when I dwitched from the olf ec_curves with separate "extra" table to the collection of individual tables, and recommend that you do the same.  You still need to have in one main table the columns you want to include i the mani browse/search, but when filling the data for one number field's web page it is easy to get other information from other tables.

Right now the onlt tables with extrs are
sage: [t for t in db.tablenames if db[t].extra_table]                      
['ec_curves',
 'ec_temp',
 'maass_newforms',
 'nf_fields',
 'nf_fields_test',
 'test_xyz']
of which ec_curves is obsolete and can be deleted (along with its extra table), ec_temp and I presume nf_fields_test and test_xyz are temporary things, so if you are able to do this for nf_fields then that will only leave maass_newforms.

**Comment by jwj61** [comment]
FYI, this is underway.  I found that I need to recompute local algebra data for some fields, so it will take a while.

**Comment by jwj61** [comment]
First, please upload nf_fields to the cloud.  I think you can just upload the main table and not the extras (if doing a rewrite preserves the links between them).  Then restart prod since this table has a new column.

Data has been computed and coding is done.  I just need to polish up the new knowls before making the pull request.

In doing this, I decided that the local algebras should be improved.  Currently, if a local algebra contains a field not in the local database, we just say "data not computed" for that prime.  That will make some search results incomplete (e.g., if Q_p is one of the completions, but we don't have all of them).

Next steps will be to recompute all of the local algebra data which preserves local factors we have, and provide partial information (e.g., degree, e, f, c) for the ones we don't have in the padic database.

**Comment by AndrewVSutherland** [comment]
nf_fields upload is starting now, I'll post again when it finishes

**Comment by AndrewVSutherland** [comment]
The upload finished, @jwj61 can you please see if things look okay?.  @roed314 there was an error complaining about the column "table_description" missing from meta_tables_hist.  I think the upload succeeded in spite of this.  I added the columns col_description and table_description to meta_tables_hist on the production database, but can you please double check and let me know if there are any other columns missing from any meta tables?

**Comment by jwj61** [comment]
Looks ok (although the new data is not being used by the code on the production server).

**Comment by roed314** [comment]
I fixed the types of `col_description` and `table_description` on `proddb` and it looks okay to me.

**Comment by AndrewVSutherland** [comment]
The latest round of changes is now live on www.lmfdb.org and seems to be working.  Should I update the tables again?

Is there anything more to do on this issue?

**Comment by jwj61** [comment]
There is still a pull request related to this.  Some data on beta breaks the number field pages for around 5 fields so that it can demo changes in that pull request.  If you upload data to the cloud, you are just breaking the corresponding pages there.

Once that pull request is merged, I have more data with the new format to insert.  When code from that pull request goes to prod, then nf_fields and lf_fields can go to the cloud.

**Comment by AndrewVSutherland** [comment]
@jwj61 Sorry, I spoke too soon, I thought #4615 had been merged, but now it has and is running on www.lmfdb.org.  I am starting the copy of lf_fields and nf_fields now.

**Comment by AndrewVSutherland** [comment]
@jwj61 Or should I wait until you add more data?

**Comment by jwj61** [comment]
Wait, now that you merged, I am adding data to nf_fields.

Copying lf_fields is fine

**Comment by AndrewVSutherland** [comment]
Okay lf_fields is on its way, let me know when nf_fields is ready.

**Comment by jwj61** [comment]
Now would be a reasonable time to upload nf_fields

**Comment by AndrewVSutherland** [comment]
nf_fields is in the process of being uploaded.

**Comment by AndrewVSutherland** [comment]
nf_fields has been copied, please let me know if it looks okay.

**Comment by jwj61** [comment]
Looks good
